panic on evans call when using reflection

## Describe the bug
I have a server with reflection, when I call some of the service methods on both 0.10.7 I get a (panic dump attached). If I use evans 0.10.6 everything works fine.

## To reproduce

- Server that has reflection turned on
- Using evans with repl mode and reflection turn on... no TLS 
- Set a header (the calls that work did not have a header but I doubt that is the problem)
- Make a call   

## Expected behavior
Call works on older release but fails with a nil ptr dereference

## Screenshots
This is what is dumped in the screen
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xa17820]

goroutine 1 [running]:
google.golang.org/protobuf/types/dynamicpb.(*Message).Descriptor(0xc908e0)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/protobuf@v1.28.0/types/dynamicpb/dynamic.go:109
google.golang.org/protobuf/internal/impl.Export.MessageDescriptorOf({}, {0xc908e0, 0x0})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/protobuf@v1.28.0/internal/impl/api_export.go:156 +0x8e
github.com/golang/protobuf/proto.MessageName({0x7f4f027fa438, 0x0})
	/Users/ktr/.ghq/pkg/mod/github.com/golang/protobuf@v1.5.2/proto/registry.go:271 +0x67
github.com/jhump/protoreflect/dynamic.isWellKnownType({0xdfe810, 0xc908e0})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/message_factory.go:167 +0x79
github.com/jhump/protoreflect/dynamic.(*KnownTypeRegistry).GetKnownType(0xb0, {0xc0002d5880, 0x7f4f028e88f0})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/message_factory.go:181 +0x165
github.com/jhump/protoreflect/dynamic.(*KnownTypeRegistry).CreateIfKnown(0xc0002a5340, {0xc0002d5880, 0xc000407450})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/message_factory.go:150 +0x25
github.com/jhump/protoreflect/dynamic.(*MessageFactory).NewMessage(0x0, 0xc0001203c0)
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/message_factory.go:64 +0x4b
github.com/jhump/protoreflect/codec.DecodeLengthDelimitedField(0xc000142bb0, {0xc00019c2c5, 0xbb4f40, 0xa0}, {0xdd0920, 0x0})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/codec/decode_fields.go:195 +0xf6
github.com/jhump/protoreflect/codec.(*Buffer).decodeKnownField(0x8, 0xc00011d680, 0xc0, {0xdd0920, 0x0})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/codec/decode_fields.go:270 +0x113
github.com/jhump/protoreflect/codec.(*Buffer).DecodeFieldValue(0xc000142bb0, 0xc000142af8, {0xdd0920, 0x0})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/codec/decode_fields.go:109 +0x91
github.com/jhump/protoreflect/dynamic.(*Message).unmarshal(0xc000160cc0, 0xc000b44bb0, 0x0)
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/binary.go:154 +0x99
github.com/jhump/protoreflect/dynamic.(*Message).UnmarshalMerge(...)
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/binary.go:149
github.com/jhump/protoreflect/dynamic.(*Message).Unmarshal(0xc0003a76e0, {0xc00019c2c0, 0xa5, 0xa5})
	/Users/ktr/.ghq/pkg/mod/github.com/jhump/protoreflect@v1.12.0/dynamic/binary.go:138 +0x8c
google.golang.org/protobuf/internal/impl.legacyUnmarshal({{}, {0xdfa598, 0xc0003a76e0}, {0xc00019c2c0, 0xa5, 0xa5}, 0x0, {0xddd3f8, 0xc0000b78f0}, 0x2710})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/protobuf@v1.28.0/internal/impl/legacy_message.go:419 +0x14a
google.golang.org/protobuf/proto.UnmarshalOptions.unmarshal({{}, 0x1, 0x1, 0x0, {0xddd3f8, 0xc0000b78f0}, 0x2710}, {0xc00019c2c0, 0xa5, 0xa5}, ...)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/protobuf@v1.28.0/proto/decode.go:104 +0x223
google.golang.org/protobuf/proto.UnmarshalOptions.UnmarshalState({{}, 0x1, 0x1, 0x0, {0x0, 0x0}, 0x2710}, {{}, {0xdfa598, 0xc0003a76e0}, ...})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/protobuf@v1.28.0/proto/decode.go:76 +0xad
github.com/golang/protobuf/proto.UnmarshalMerge({0xc00019c2c0, 0xa5, 0xa5}, {0xdde160, 0xc000160cc0})
	/Users/ktr/.ghq/pkg/mod/github.com/golang/protobuf@v1.5.2/proto/wire.go:67 +0x149
github.com/golang/protobuf/proto.Unmarshal({0xc00019c2c0, 0xa5, 0xa5}, {0xdde160, 0xc000160cc0})
	/Users/ktr/.ghq/pkg/mod/github.com/golang/protobuf@v1.5.2/proto/wire.go:58 +0x65
google.golang.org/grpc/encoding/proto.codec.Unmarshal({}, {0xc00019c2c0, 0xa5, 0xa5}, {0xcb9420, 0xc000160cc0})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/encoding/proto/proto.go:53 +0x65
google.golang.org/grpc.recv(0xc000412378, {0x7f4f02b4a320, 0x13aa308}, 0xc000932e40, {0x0, 0x0}, {0xcb9420, 0xc000160cc0}, 0x976269, 0x0, ...)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/rpc_util.go:760 +0xb5
google.golang.org/grpc.(*csAttempt).recvMsg(0xc00058ae70, {0xcb9420, 0xc000160cc0}, 0x0)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/stream.go:983 +0x2b0
google.golang.org/grpc.(*clientStream).RecvMsg.func1(0x20)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/stream.go:834 +0x25
google.golang.org/grpc.(*clientStream).withRetry(0xc000512360, 0xc000b451a0, 0xc000b45170)
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/stream.go:692 +0xd3
google.golang.org/grpc.(*clientStream).RecvMsg(0xc000512360, {0xcb9420, 0xc000160cc0})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/stream.go:833 +0x11f
google.golang.org/grpc.invoke({0xde45d0, 0xc000160c90}, {0xc00003ed40, 0x20}, {0xcb9420, 0xc000932e40}, {0xcb9420, 0xc000160cc0}, 0x0, {0xc00050b820, ...})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/call.go:73 +0xd7
google.golang.org/grpc.(*ClientConn).Invoke(0x7f4f298975b8, {0xde45d0, 0xc000160c90}, {0xc00003ed40, 0x0}, {0xcb9420, 0xc000932e40}, {0xcb9420, 0xc000160cc0}, {0xc00050b820, ...})
	/Users/ktr/.ghq/pkg/mod/google.golang.org/grpc@v1.46.0/call.go:37 +0x265
github.com/ktr0731/evans/grpc.(*client).Invoke(0xc000405660, {0xde45d0, 0xc000160c90}, {0xc000200330, 0x30}, {0xcb9420, 0xc000932e40}, {0xcb9420, 0xc000160cc0})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/grpc/grpc.go:188 +0x255
github.com/ktr0731/evans/usecase.(*dependencyManager).CallRPC(0xc000376d00, {0xde4560, 0xc00003a0c0}, {0x40cc94, 0xc0001436b8}, {0xc00012c2b0, 0x18}, 0x0, {0xdd0ae0, 0xc000520028})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/usecase/call_rpc.go:411 +0xf83
github.com/ktr0731/evans/usecase.(*dependencyManager).CallRPCInteractively(0xc000376d00, {0xde4560, 0xc00003a0c0}, {0xdd2520, 0xc00000e018}, {0xc00012c2b0, 0x7}, 0x0, 0x0, 0x0, ...)
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/usecase/call_rpc.go:486 +0x134
github.com/ktr0731/evans/usecase.CallRPCInteractively(...)
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/usecase/call_rpc.go:482
github.com/ktr0731/evans/repl.(*callCommand).Run(0xc0002eb100, {0xdd2520, 0xc00000e018}, {0xc0003a6b10, 0x1, 0x4})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/repl/commands.go:204 +0x332
github.com/ktr0731/evans/repl.(*REPL).runCommand(0xc00051af00, {0xc00012c290, 0x4}, {0xc000396650, 0x1, 0x1})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/repl/repl.go:156 +0x274
github.com/ktr0731/evans/repl.(*REPL).Run(0xc00051af00, {0xdebcb8, 0xc0000fee00})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/repl/repl.go:111 +0x22f
github.com/ktr0731/evans/mode.RunAsREPLMode(0xc000335ad0, {0xdeae98, 0xc0004468f0}, 0xc000414140)
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/mode/repl.go:68 +0x830
github.com/ktr0731/evans/app.runREPLCommand(0xc000403080, {0xdeae98, 0xc0004468f0})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/app/commands.go:280 +0x2e5
github.com/ktr0731/evans/app.newREPLCommand.func1(0xc0001f5ce0, 0xc000403080)
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/app/commands.go:248 +0xd1
github.com/ktr0731/evans/app.runFunc.func1(0xc00038b180, {0xc000267590, 0x0, 0x5})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/app/commands.go:87 +0x163
github.com/spf13/cobra.(*Command).execute(0xc00038b180, {0xc000267540, 0x5, 0x5})
	/Users/ktr/.ghq/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0xc00038a500)
	/Users/ktr/.ghq/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/ktr/.ghq/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/ktr0731/evans/app.(*App).Run(0xc000b45f58, {0xc000030080, 0x6, 0x60})
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/app/app.go:51 +0x94
main.main()
	/Users/ktr/.ghq/src/github.com/ktr0731/evans/main.go:11 +0xb3

## Environment
- OS: Ubuntu 22.0.4
uname -a
Linux XXXXXX #43-Ubuntu SMP Wed Jun 15 12:54:21 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
Using the amd64 linux tar

- Terminal: GNOME terminal 3.44.0 using zsh and bash (makes no difference)
- Evans version: 0.10.7
- `protoc` version: protoc --version libprotoc 3.19.4
- `protoc` plugin version (if you are using):

## Additional context
If I use evans 0.10.6 everything works fine. 
At first I thought it was reflection but even passing the protos directly causes the same issue. 
The service is implemented in go 1.18.2 but the same instance works for 0.10.6 and not 0.10.7

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -31,8 +31,7 @@ jobs:
       run: go vet ./...
 
     - name: Test
-      # TODO: Remove conflictPolicy flag.
-      run: go test -race -coverpkg ./... -covermode atomic -coverprofile coverage.txt -ldflags "-X google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn" ./...
+      run: go test -p 1 -coverpkg ./... -covermode atomic -coverprofile coverage.txt ./...
 
     - name: Upload coverage to Codecov
       uses: codecov/codecov-action@v3.1.0

--- a/Makefile
+++ b/Makefile
@@ -32,8 +32,7 @@ credits:
 
 .PHONY: gotest
 gotest: lint
-	# TODO: Remove conflictPolicy flag.
-	go test -race -ldflags "-X google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn" ./...
+	go test -race ./...
 
 .PHONY: lint
 lint:

--- a/idl/proto/proto.go
+++ b/idl/proto/proto.go
@@ -14,9 +14,6 @@ import (
 	"github.com/ktr0731/evans/grpc/grpcreflection"
 	"github.com/ktr0731/evans/idl"
 	"github.com/pkg/errors"
-	"google.golang.org/protobuf/reflect/protodesc"
-	"google.golang.org/protobuf/reflect/protoregistry"
-	"google.golang.org/protobuf/types/dynamicpb"
 )
 
 type spec struct {
@@ -138,7 +135,12 @@ func LoadFiles(importPaths []string, fnames []string) (idl.Spec, error) {
 		return nil, errors.Wrap(err, "proto: failed to parse passed proto files")
 	}
 
-	return newSpec(fileDescs)
+	// Collect dependency file descriptors
+	for _, d := range fileDescs {
+		fileDescs = append(fileDescs, d.GetDependencies()...)
+	}
+
+	return newSpec(fileDescs), nil
 }
 
 // LoadByReflection receives a gRPC reflection client, then tries to instantiate a new idl.Spec by using gRPC reflection.
@@ -147,10 +149,10 @@ func LoadByReflection(client grpcreflection.Client) (idl.Spec, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to list packages by gRPC reflection")
 	}
-	return newSpec(fileDescs)
+	return newSpec(fileDescs), nil
 }
 
-func newSpec(fds []*desc.FileDescriptor) (idl.Spec, error) {
+func newSpec(fds []*desc.FileDescriptor) idl.Spec {
 	var (
 		encounteredPkgs = make(map[string]interface{})
 		encounteredSvcs = make(map[string]interface{})
@@ -160,10 +162,6 @@ func newSpec(fds []*desc.FileDescriptor) (idl.Spec, error) {
 		msgDescs        = make(map[string]*desc.MessageDescriptor)
 	)
 	for _, f := range fds {
-		if err := registerFileAndType(f); err != nil {
-			return nil, errors.Wrap(err, "failed to register file descriptor")
-		}
-
 		if _, encountered := encounteredPkgs[f.GetPackage()]; !encountered {
 			pkgNames = append(pkgNames, f.GetPackage())
 			encounteredPkgs[f.GetPackage()] = nil
@@ -192,38 +190,7 @@ func newSpec(fds []*desc.FileDescriptor) (idl.Spec, error) {
 		svcDescs:  svcDescs,
 		rpcDescs:  rpcDescs,
 		msgDescs:  msgDescs,
-	}, nil
-}
-
-func registerFileAndType(fd *desc.FileDescriptor) error {
-	deps := fd.GetDependencies()
-	for _, d := range deps {
-		if err := registerFileAndType(d); err != nil {
-			return err
-		}
 	}
-
-	prfd, err := protodesc.NewFile(fd.AsFileDescriptorProto(), protoregistry.GlobalFiles)
-	if err != nil {
-		return errors.Wrap(err, "failed to new file descriptor")
-	}
-
-	if _, err := protoregistry.GlobalFiles.FindFileByPath(prfd.Path()); errors.Is(err, protoregistry.NotFound) {
-		if err := protoregistry.GlobalFiles.RegisterFile(prfd); err != nil {
-			return err
-		}
-	}
-
-	for i := 0; i < prfd.Messages().Len(); i++ {
-		md := prfd.Messages().Get(i)
-		if _, err := protoregistry.GlobalTypes.FindMessageByName(md.FullName()); errors.Is(err, protoregistry.NotFound) {
-			if err := protoregistry.GlobalTypes.RegisterMessage(dynamicpb.NewMessageType(md)); err != nil {
-				return err
-			}
-		}
-	}
-
-	return nil
 }
 
 // FullyQualifiedServiceName returns the fully-qualified service name.

